clients/java/signalr/src/main/java/com/microsoft/aspnet/signalr/HubConnection.java

```diff
@@ -132,6 +138,14 @@ public HubConnection(String url, HttpConnectionOptions options) {
         };
     }
 
+    private <T> CompletableFuture<T> timeoutAfter(long timeout, TimeUnit unit) {
+        CompletableFuture<T> result = new CompletableFuture<T>();
+        ScheduledExecutorService scheduledThreadPool = Executors.newScheduledThreadPool(5);
```

Comment:
Why not `newSingleThreadScheduledExecutor`

Reply:
That would probably be better